### PR TITLE
creating-services

### PR DESCRIPTION
**Pull request by nbsven**

### CODE REVIEW

File: src/main/java/com/epam/magazinestore/services/PaymentService.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ */
+public interface PaymentService {
+
+    /**
+     * Register new {@code Payment} in database with {@code false} in paid field.
+     *
+     * @param payment {@code Payment} that need to be registered.
+     * @return {@code true} if paid was successfully registered.
+     */
+    boolean registerPayment(Payment payment);
```
Comment (Jebsuz) [inline src/main/java/com/epam/magazinestore/services/PaymentService.java]: who will construct this `Payment`? Maybe `PaymentService` will accept an `Subscription` as parameter and therefore will create a `Payment` for it?

File: src/main/java/com/epam/magazinestore/services/PaymentService.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    /**
+     * Accept {@code Payment} (make payment paid).
+     *
+     * @param payment {@code Payment} that needs to be accepted.
+     * @return {@code true} if payment was paid.
+     */
+    boolean acceptPayment(Payment payment);
+
+
+}
```
Comment (Jebsuz) [inline src/main/java/com/epam/magazinestore/services/PaymentService.java]: how do you plan to do a page with all payments for administrator if there is not API to get the `Payment`?

File: src/main/java/com/epam/magazinestore/services/SubscriptionService.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+    /**
+     * Initialize new {@code Subscription}, if subscription not exists.
+     *
+     * @param account  {@code Account}, that need to be subscribed.
+     * @param magazine for subscription.
+     * @param duration Months of subscription.
+     * @return {@code null} if initializing was not success, and the created {@code Subscription} in other cases.
+     */
+    Subscription subscribe(Account account, Magazine magazine, int duration);
```
Comment (Jebsuz) [inline src/main/java/com/epam/magazinestore/services/SubscriptionService.java]: what is a duration, months, days, hours, years? be more specific when you name fields/params/variables
Yes, you did provide a javadoc, but if you would name duration as `durationInMonths` or `months` then i would no longer need to read a javadoc, the code would speak for itself

Comment (Jebsuz) [inline src/main/java/com/epam/magazinestore/services/SubscriptionService.java]: Why package-private?

Comment (Jebsuz) [inline src/main/java/com/epam/magazinestore/services/SubscriptionService.java]: In other services you used the 'register' prefix for methods that adds something, why here you named just `subscribe`? Be more consistent with names across your application. With names you form some kinda 'convention' inside your application with consistent name strategies so other people would navigate and determine what method does faster

File: src/main/java/com/epam/magazinestore/services/SubscriptionService.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+    /**
+     * Extend already existing {@code Subscription} for duration (in months)
+     *
+     * @param account  {@code Account} for extend subscription.
+     * @param magazine {@code Magazine} for subscription.
+     * @param duration Months of subscription extending.
+     * @return {@code true} if extending was success.
+     */
+    boolean extendSubsription(Account account, Magazine magazine, int duration);
```
Comment (Jebsuz) [inline src/main/java/com/epam/magazinestore/services/SubscriptionService.java]: if we are about to extend an existing subscription, why we must provide so much params? `Subscription` itself and and a new duration is enough imo

File: src/main/java/com/epam/magazinestore/services/SubscriptionService.java (hunk 3 above)
Comment (vakhonin) [reply src/main/java/com/epam/magazinestore/services/SubscriptionService.java]: In interfaces, all methods are public, aren't they ?

Comment (Jebsuz) [reply src/main/java/com/epam/magazinestore/services/SubscriptionService.java]: ugh, my bad, github's color scheme confused me, sorry

**Pull request merged**